packages/flutter/lib/src/painting/continuous_rectangle_border.dart

```diff
@@ -3,58 +3,276 @@
+        ..lineTo(rightX(0.07491100), bottomY(0.63149399))
+        ..cubicTo(rightX(0.16905899), bottomY(0.37282392),
+            rightX(0.37282401), bottomY(0.16906013),
+            rightX(0.63149399), bottomY(0.07491176))
+        ..cubicTo(rightX(0.86840701), bottomY(0),
+            rightX(1.08849299), bottomY(0.0),
+            rightX(1.52866483), bottomY(0.0))
+        ..lineTo(leftX(1.52866483), bottomY(0.0))
+        ..cubicTo(leftX(1.08849323), bottomY(0.0),
+            leftX(0.86840689), bottomY(0.0),
+            leftX(0.66993427), bottomY(0.06549600))
+        ..lineTo(leftX(0.63149399), bottomY(0.07491100))
+        ..cubicTo(leftX(0.37282392), bottomY(0.16905899),
+            leftX(0.16906013), bottomY(0.37282401),
+            leftX(0.07491176), bottomY(0.63149399))
+        ..cubicTo(leftX(0), bottomY(0.86840701),
+            leftX(0.0), bottomY(1.08849299),
+            leftX(0.0), bottomY(1.52866483))
+        ..lineTo(leftX(0.0), topY(1.52866471))
+        ..cubicTo(leftX(0.0), topY(1.08849323),
+            leftX(0.0), topY(0.86840689),
+            leftX(0.06549600), topY(0.66993427))
+        ..lineTo(leftX(0.07491100), topY(0.63149399))
+        ..cubicTo(leftX(0.16905899), topY(0.37282392),
+            leftX(0.37282401), topY(0.16906013),
+            leftX(0.63149399), topY(0.07491176))
+        ..cubicTo(leftX(0.86840701), topY(0),
+            leftX(1.08849299), topY(0.0),
+            leftX(1.52866483), topY(0.0))
+        ..close();
+    }
+
+    // The ratio of the declared corner radius to the total affected pixels to
+    // render the corner. For example if the declared radius were 25px then
+    // totalAffectedCornerPixelRatio * 25 (~38) pixels would be affected.
+    const double totalAffectedCornerPixelRatio = 1.52865;
+
+    // The radius multiplier where the resulting shape will concave with a
+    // height and width of any value.
+    //
+    // If the shortest side length to radius ratio drops below this value, the
+    // radius must be lessened to avoid clipping (ie. concavity) of the shape.
+    //
+    // This value comes from the website where the other equations and curves
+    // were found
+    // (https://www.paintcodeapp.com/news/code-for-ios-7-rounded-rectangles),
+    // however it represents the ratio of the total 90º curve width or height to
+    // the width or height of the smallest rectangle dimension.
+    const double minimalUnclippedSideToCornerRadiusRatio = 2.0 * totalAffectedCornerPixelRatio;
+
+    // The multiplier of the radius in comparison to the smallest edge length
+    // used to describe the minimum radius for this shape.
+    //
+    // This is multiplier used in the case of an extreme aspect ratio and a
+    // small extent value. It can be less than 'maxMultiplier' because there
+    // are not enough pixels to render the clipping of the shape at this size so
+    // it appears to still be concave (whereas mathematically it's convex).
```

Comment:
I'm still confused about these various usages of concave vs convex. Isn't the shape always convex? When does it become concave? Or are you describing the path drawing glitch caused by this specific implementation of the bezier curve when it clips?

Reply:
I now realize that a lozenge isn't actually convex. But yah a lozenge doesn't correctly describe the shape that the curves make when they clip as the shape in reality is more like |X| with a lozenge in middle. Which is a convex shape.

Reply:
I think we're mixing a bunch of things.

If we're talking about these shapes assuming we produce protruding rendering artifacts when the shapes clip, then:
- If it doesn't clip, it's a simple non-intersecting polygon that is convex
- If it does clip, it is a non-simple self-intersecting polygon (which by definition is neither convex nor concave).
We shouldn't do this since the fact that we produce rendering artifacts is an implementation detail that we use bezier curves and has nothing to do with superellipses themselves. 

If we're talking about these shapes assuming we don't render protruding artifacts (e.g. we end up producing a lozenge), then:
- If it doesn't clip (n>2), it's a simple non-intersecting polygon that is convex.
- If it does clip (n<2), then it is still a simple non-intersecting polygon that is convex